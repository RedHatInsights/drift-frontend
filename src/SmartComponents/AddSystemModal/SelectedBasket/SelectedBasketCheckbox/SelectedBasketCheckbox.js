import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from '@patternfly/react-core';

export class SelectedBasketCheckbox extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isChecked: true
        };
    }

    handleChange = () => {
        const { findType, id, type } = this.props;
        const { isChecked } = this.state;

        findType(type, id);
        this.setState({ isChecked: !isChecked });
    }

    render() {
        const { isChecked } = this.state;

        return (
            <React.Fragment>
                <Checkbox
                    isChecked={ isChecked }
                    onChange={ this.handleChange }
                    aria-label="controlled checkbox example"
                    id="check-1"
                    name="check1"
                />
            </React.Fragment>
        );
    }
}

SelectedBasketCheckbox.propTypes = {
    type: PropTypes.string,
    findType: PropTypes.func,
    id: PropTypes.string
};

export default SelectedBasketCheckbox;
