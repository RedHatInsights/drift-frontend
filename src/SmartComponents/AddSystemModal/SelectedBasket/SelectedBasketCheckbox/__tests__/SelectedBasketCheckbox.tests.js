/*eslint-disable camelcase*/
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SelectedBasketCheckbox from '../SelectedBasketCheckbox';

describe('SelectedBasketCheckbox', () => {
    let props;

    beforeEach(() => {
        props = {
            type: '',
            id: 'abcd1234',
            findType: jest.fn()
        };
    });

    it('should render system', () => {
        props.type = 'system';

        const wrapper = shallow(
            <SelectedBasketCheckbox
                { ...props }
            />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render baseline', () => {
        props.type = 'baseline';

        const wrapper = shallow(
            <SelectedBasketCheckbox
                { ...props }
            />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render hsp', () => {
        props.type = 'hsp';

        const wrapper = shallow(
            <SelectedBasketCheckbox
                { ...props }
            />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should handle change', () => {
        props.type = 'system';

        const wrapper = shallow(
            <SelectedBasketCheckbox
                { ...props }
            />
        );

        wrapper.instance().handleChange();

        expect(props.findType).toHaveBeenCalledWith('system', 'abcd1234');
        expect(wrapper.state('isChecked')).toEqual(false);
    });
});
/*eslint-enable camelcase*/
