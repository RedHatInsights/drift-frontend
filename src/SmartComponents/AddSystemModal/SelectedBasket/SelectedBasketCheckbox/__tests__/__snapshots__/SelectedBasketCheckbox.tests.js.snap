// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedBasketCheckbox should render baseline 1`] = `
<Fragment>
  <Checkbox
    aria-label="controlled checkbox example"
    className=""
    component="div"
    id="check-1"
    isChecked={true}
    isDisabled={false}
    isValid={true}
    name="check1"
    onChange={[Function]}
    ouiaSafe={true}
  />
</Fragment>
`;

exports[`SelectedBasketCheckbox should render hsp 1`] = `
<Fragment>
  <Checkbox
    aria-label="controlled checkbox example"
    className=""
    component="div"
    id="check-1"
    isChecked={true}
    isDisabled={false}
    isValid={true}
    name="check1"
    onChange={[Function]}
    ouiaSafe={true}
  />
</Fragment>
`;

exports[`SelectedBasketCheckbox should render system 1`] = `
<Fragment>
  <Checkbox
    aria-label="controlled checkbox example"
    className=""
    component="div"
    id="check-1"
    isChecked={true}
    isDisabled={false}
    isValid={true}
    name="check1"
    onChange={[Function]}
    ouiaSafe={true}
  />
</Fragment>
`;
